### PR TITLE
Added DecodeNoLimitGeneric to bech32.go

### PR DESCRIPTION
**Pull request by MarcoEzekiel**
a fix for issue  Public DecodeNoLimit(encoded) #2185

### CODE REVIEW

File: btcutil/bech32/bech32.go [hunk 1]
```diff
@@ -266,6 +266,15 @@ func DecodeNoLimit(bech string) (string, []byte, error) {
 	return hrp, data, err
 }
 
+// DecodeGeneric is identical to the existing DecodeNoLimit method, but will also
+// return the bech32 version that matches the decoded checksum.
+//
+// Note that the returned data is 5-bit (base32) encoded and the human-readable
+// part will be lowercase.
+func DecodeNoLimitGeneric(bech string) (string, []byte, Version, error) {
+	return decodeNoLimit(bech)
```
Comment (guggero) [inline btcutil/bech32/bech32.go:275]: How about we just export the `decodeNoLimit` and re-name it to `DecodeNoLimitWithVersion`? "Generic" doesn't really say that much IMO (at least in this context).

Comment (MarcoEzekiel) [reply btcutil/bech32/bech32.go:275]: I may have misunderstood your comment, are you suggesting renaming the private method decodeNoLimit  as a Public method named DecodeNoLimitWithVersion  and changing references to method in the package?

Comment (MarcoEzekiel) [reply btcutil/bech32/bech32.go]: Done!

**Review by guggero — COMMENTED** [review]
Code looks good to me. Can you please squash the two commits and add the `btcutil/bech32: ` prefix to the commit message?

File: btcutil/bech32/bech32.go (hunk 1 above)
Comment (MarcoEzekiel) [reply btcutil/bech32/bech32.go]: I have cleaned this up as requested.

**Review by guggero — APPROVED** [review]
Thank you, LGTM :tada:

**Comment by leandrogomesmachado** [comment]
❤️ Good job

**Comment by MarcoEzekiel** [comment]
Any Chance this will be merged soon?

**Pull request merged**